Fix Checkstyle Errors

heidijinxujia opened the pull request:
RFC #92 
Current there are **743** errors in total to fix.

I am going to fix them in the sequence of folders.

0/743 errors left. (May 17th. 💯 !)

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/adal/internal/net/HttpWebRequest.java:
```diff
@@ -161,6 +180,15 @@ public HttpWebResponse send() throws IOException {
+    /**
+     * Check if the network is available. If the network is unavailable, {@link ClientException}
+     * will throw with error code {@link ErrorStrings#NO_NETWORK_CONNECTION_POWER_OPTIMIZATION}
+     * when Connection is not available to refresh token because power optimization is enabled, or
```
nit: connection

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/adal/internal/util/DateExtensions.java:
```diff
@@ -34,6 +34,9 @@ private DateExtensions() {
      * Create an immutable object for the input Date object
      * to avoid exposing the internal references.
+     *
+     * @param date Date
```
do you absolutely need to write something? Date Date looks a little weird.

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/adal/internal/util/HashMapExtensions.java:
```diff
@@ -99,7 +99,7 @@ private HashMapExtensions() {
      * @param webResponse HttpWebResponse to convert to a map
      * @return Map
-     * @throws JSONException
+     * @throws JSONException throws if the response body is JSON malformed.
```
is malformed

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/adal/internal/util/JsonExtensions.java:
```diff
@@ -30,10 +30,19 @@
+     *
+     * @param jsonString String
+     * @return Map<String, String>
+     * @throws JSONException throws if string is JSON malformed.
```
if the JSON is malformed

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/adal/internal/util/StringExtensions.java:
```diff
@@ -138,6 +159,13 @@ public static URL getUrl(String endpoint) {
     }
 
+    /**
+     * Get string list split with delimiter.
```
can we change this string

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/adal/internal/util/StringExtensions.java:
```diff
@@ -198,6 +239,12 @@ public static boolean hasPrefixInHeader(final String value, final String prefix)
     }
 
+    /**
+     * Get the Base64 encode to string of message.
```
Based64URL encode the input string

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/exception/ErrorStrings.java:
```diff
@@ -22,7 +22,9 @@
 
-public class ErrorStrings {
+public final class ErrorStrings {
+    private ErrorStrings() {
```
add a line before the constructor

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/internal/cache/ICacheKeyValueDelegate.java:
```diff
@@ -27,15 +27,46 @@
+     * Get the account credential from cache value.
+     *
+     * @param string String
+     * @param t      AccountCredentialBase
```
extra tab

heidijinxujia replied:
This is not a tab. This is auto alignment for the Javadoc.

Thank you for the CR @addev-ashish, I have addressed the rest as well as all other occurrences in the code.

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/internal/dto/AccessToken.java:
```diff
@@ -34,11 +34,29 @@
+         * String of access token type.
+         */
         public static final String ACCESS_TOKEN_TYPE = "access_token_type";
+        /**
```
a line before the comments here please

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationRequest.java:
```diff
@@ -36,44 +36,74 @@
     private String mState;
 
+    /**
+     * @return mResponseType of the authorization request
```
.

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/AuthorizationRequest.java:
```diff
@@ -36,44 +36,74 @@
-    public void setResponseType(String mResponseType) {
-        this.mResponseType = mResponseType;
+    /**
+     * @param responseType response type of the authorization request
```
.

addev-ashish replied:
everywhere in this file

addev-ashish commented on common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenErrorResponse.java:
```diff
@@ -30,46 +30,69 @@
+    private String mCorrelationId;
 
+    /**
+     * @return mErrorCodes of the Microsoft token error response
```
. in this file

heidijinxujia commented:
Thank you for the CR @addev-ashish, I have addressed the rest as well as all other occurrences in the code.

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:
```diff
@@ -56,9 +56,9 @@
 public class ADALOAuth2TokenCache
```
Depending on which merges first, there are some conflicts to reconcile between #96 and #103

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2TokenCache.java:
```diff
@@ -54,4 +53,12 @@ public OAuth2TokenCache(Context context) {
                                     final U request,
                                     final V response) throws ClientException;
+
+    protected Context getContext() {
```
This should be removed. Doesn't make sense to expose this as API

heidijinxujia replied:
mContext has be private and needs it own getter and setter based on Checkstyle's warning. mContext gets called in ADALOAuth2TokenCache. This is reason I add this getter as protected method here.

heidijinxujia replied:
mContext has be private and needs it own getter and setter based on Checkstyle's warning. mContext gets called in ADALOAuth2TokenCache. This is reason I add this getter as protected method here.

`[ant:checkstyle] [WARN] ..\common\src\main\java\com\microsoft\identity\common\internal\providers\oauth2\OAuth2TokenCache.java:34:29: Variable 'mContext' must be private and have accessor methods. [VisibilityModifier]`

iambmelt replied:
👍

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2TokenCache.java:
```diff
@@ -54,4 +53,12 @@ public OAuth2TokenCache(Context context) {
+        return mContext;
+    }
+
+    protected void setContext(Context context) {
```
Same here. Shouldn't be part of the `OAuth2TokenCache` API

heidijinxujia replied:
See above.

iambmelt replied:
Let's remove this and declare `private final`

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java:
```diff
@@ -105,59 +115,93 @@ protected HttpResponse performTokenRequest(final GenericTokenRequest request) th
         return authorizationUri;
     }
 
+    protected String getAuthorizationEndpoint() {
```
I'm not sure about the access modifiers of these methods -- I suspect one of a few things might happen in the future:

1. We'll make these `public`
2. We'll declare these fields `final`, initialize them in the constructor, and provider `public` getters _only_

No action required for this. Mostly a mental note. Tagging @shoatman as FYI

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java:
```diff
@@ -105,59 +115,93 @@ protected HttpResponse performTokenRequest(final GenericTokenRequest request) th
+        return mAuthorizationEndpoint;
+    }
+
+    protected void setAuthorizationEndpoint(String authorizationEndpoint) {
```
Could mark params final but I'm not going to stress it since these might change.

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java:
```diff
@@ -105,59 +115,93 @@ protected HttpResponse performTokenRequest(final GenericTokenRequest request) th
+    protected void setTokenEndpoint(String tokenEndpoint) {
+        this.mTokenEndpoint = tokenEndpoint;
+    }
+
```
Could mark params final but I'm not going to stress it since these might change.

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java:
```diff
@@ -105,59 +115,93 @@ protected HttpResponse performTokenRequest(final GenericTokenRequest request) th
+        return mIssuer;
+    }
+
+    protected void setIssuer(Uri issuer) {
```
Could mark params final but I'm not going to stress it since these might change.

iambmelt commented:
Lots of places where parameters are not declared as `final` -- as this is something we've made a bit of a habit out of, let's take care of those for the accessors added in this PR. I won't comment each individual one

iambmelt commented:
@heidijinxujia Can you add your name and email to your `.gitconfig`? All of your commits show the author as `Unknown`

![capture](https://user-images.githubusercontent.com/990063/40240644-05ec1fdc-5a87-11e8-9a14-f4bb0c681687.JPG)

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:
```diff
@@ -86,7 +104,7 @@ public ADALOAuth2TokenCache(final Context context) {
 
     protected void initializeSharedPreferencesFileManager(final String fileName) {
-        mISharedPreferencesFileManager = new SharedPreferencesFileManager(super.mContext, fileName);
+        setISharedPreferencesFileManager(new SharedPreferencesFileManager(super.getContext(), fileName));
```
Remove `getContext` method, declare as `final protected`

heidijinxujia replied:
Checkstyle won't let us do this way. See above.

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:
```diff
@@ -162,7 +180,7 @@ protected StorageHelper getStorageHelper() {
             if (sHelper == null) {
                 Log.v(TAG, "Started to initialize storage helper");
-                sHelper = new StorageHelper(mContext);
+                sHelper = new StorageHelper(this.getContext());
```
See above, use `mContext`

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:
```diff
@@ -69,7 +69,20 @@
+        return mISharedPreferencesFileManager;
+    }
+
+    protected void setISharedPreferencesFileManager(ISharedPreferencesFileManager iSharedPreferencesFileManager) {
```
Let's remove this and declare `ISharedPreferencesFileManager` as `final` -- initialize once in the constructor. This shouldn't be swappable.

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:
```diff
@@ -69,7 +69,20 @@
 
     private List<IShareSingleSignOnState> mSharedSSOCaches;
 
+    protected ISharedPreferencesFileManager getISharedPreferencesFileManager() {
```
Remove. Shouldn't be part of the API for this class

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:
```diff
@@ -147,7 +165,7 @@ private void setItem(String key, ADALTokenCacheItem cacheItem) {
         String encrypted = encrypt(json);
         if (encrypted != null) {
-            mISharedPreferencesFileManager.putString(key, encrypted);
+            getISharedPreferencesFileManager().putString(key, encrypted);
```
Dereference directly -- this field shouldn't have a getter

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/cache/AccountCredentialCache.java:
```diff
@@ -471,6 +484,7 @@ private CredentialType getCredentialTypeForCredentialCacheKey(final String cache
                     type = CredentialType.IdToken;
                 } else {
                     // TODO Log a warning and skip this value?
+                    Logger.verbose(TAG, "Unexpected credential type.");
```
`Logger.warn`

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/dto/CredentialType.java:
```diff
@@ -29,13 +29,41 @@
 import java.util.Set;
 
 public enum CredentialType {
+    /**
```
I'm not sure what the value of this is - thanks checkstyle 🙄 


🤷‍♂️

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAccount.java:
```diff
@@ -104,12 +114,18 @@ private String getUniqueId(Map<String, String> claims) {
+     * @param givenName given name of the Microsoft account.
+     */
+    public void setFirstName(String givenName) {
+        this.mGivenName = givenName;
```
No `this`

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftAccount.java:
```diff
@@ -104,12 +114,18 @@ private String getUniqueId(Map<String, String> claims) {
+     * @param familyName family name of the Microsoft account.
+     */
+    public void setLastName(String familyName) {
+        this.mFamilyName = familyName;
```
No `this`

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftTokenResponse.java:
```diff
@@ -26,8 +26,14 @@
+    }
 
+    protected void setExtendedExpiresIn(Long extendedExpiresIn) {
+        this.mExtendedExpiresIn = extendedExpiresIn;
```
No `this`

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryAuthorizationRequest.java:
```diff
@@ -28,14 +28,20 @@
+     * @param authority URL
+     */
+    public void setAuthority(URL authority) {
+        this.mAuthority = authority;
```
No `this`

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java:
```diff
@@ -52,22 +52,28 @@
-    protected String mTokenEndpoint;
-    protected String mAuthorizationEndpoint;
-    protected Uri mIssuer;
+    private String mTokenEndpoint;
```
As these values are used, set, and accessed by subclasses, I'd prefer to keep them `protected`

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryOAuth2Strategy.java:
```diff
@@ -54,10 +54,15 @@
     public AzureActiveDirectoryOAuth2Strategy(final AzureActiveDirectoryOAuth2Configuration config) {
         super(config);
-        mTokenEndpoint = "https://login.microsoftonline.com/microsoft.com/oauth2/token";
+        this.setTokenEndpoint("https://login.microsoftonline.com/microsoft.com/oauth2/token");
```
constructors (of non-final classes) should call only final or private methods. 

Categorically, I really dislike it when non-`final` classes has constructors which call non-`final` or non-`private` methods as @Overrides can have strange and unforeseen consequences.

Let's make `mTokenEndpoint` `protected` and avoid this entirely

iambmelt replied:
Let's make `setTokenEndpoint` a `final` method -- should keep checkstyle happy while satisfying the previous comment

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryOAuth2Strategy.java:
```diff
@@ -140,7 +145,7 @@ protected TokenResult getTokenResultFromHttpResponse(final HttpResponse response
         TokenErrorResponse tokenErrorResponse = null;
 
-        if (response.getStatusCode() >= 400) {
+        if (response.getStatusCode() >= HTTP_BAD_REQUEST) {
```
Should use `HttpURLConnection.html#HTTP_BAD_REQUEST`

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAccessToken.java:
```diff
@@ -26,10 +26,23 @@
+    }
+
+    protected void setExtExpiresIn(Long extExpiresIn) {
+        this.mExtExpiresIn = extExpiresIn;
```
No `this`

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java:
```diff
@@ -121,7 +127,7 @@ protected TokenResult getTokenResultFromHttpResponse(final HttpResponse response
         TokenErrorResponse tokenErrorResponse = null;
 
-        if (response.getStatusCode() >= 400) {
+        if (response.getStatusCode() >= HTTP_BAD_REQUEST) {
```
Should use `HttpURLConnection.html#HTTP_BAD_REQUEST`

iambmelt commented on config/checkstyle/suppressions.xml:
```diff
@@ -4,5 +4,8 @@
 
 <suppressions>
-  <!--add any suppressions here -->
+  <suppress checks=".*" files=".*Test.java"/>
```
👍

iambmelt commented on config/checkstyle/suppressions.xml:
```diff
@@ -4,5 +4,8 @@
-  <!--add any suppressions here -->
+  <suppress checks=".*" files=".*Test.java"/>
+  <suppress checks=".*" files=".*[\\/]unit[\\/]"/>
+  <suppress checks="JavadocVariable" files="MicrosoftStsIdToken.java"/>
```
What's this?

heidijinxujia replied:
Could not locate any documentation for these two variables.
`public static final String AIO = "aio";
    public static final String UTI = "uti";`

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java:
```diff
@@ -105,59 +115,93 @@ protected HttpResponse performTokenRequest(final GenericTokenRequest request) th
      */
     public abstract String getIssuerCacheIdentifier(GenericAuthorizationRequest request);
 
+    /**
```
https://github.com/AzureAD/microsoft-authentication-library-common-for-android/issues/104

heidijinxujia commented on common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2TokenCache.java:
```diff
@@ -27,12 +27,11 @@
         <T extends OAuth2Strategy, U extends AuthorizationRequest, V extends TokenResponse> {
-
-    protected Context mContext;
+    private Context mContext;
```
See above.

iambmelt replied:
👍

heidijinxujia commented:
@addev-ashish @iambmelt Thank you for the comments. Addressed. Please have a look!

heidijinxujia commented:
@iambmelt thanks for the comments. Addressed.

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/dto/IdToken.java:
```diff
@@ -84,21 +90,32 @@ public void setRealm(final String realm) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-        if (!super.equals(o)) return false;
+        if (this == o) {
```
More of this -- not a good use of our time

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/dto/IdToken.java:
```diff
@@ -84,21 +90,32 @@ public void setRealm(final String realm) {
         int result = super.hashCode();
-        result = 31 * result + (mRealm != null ? mRealm.hashCode() : 0);
-        result = 31 * result + (mAuthority != null ? mAuthority.hashCode() : 0);
+        result = UNIQUE_ID_LENGTH * result + (mRealm != null ? mRealm.hashCode() : 0);
```
Same here. I won't comment on more of these.

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/MicrosoftClientAssertion.java:
```diff
@@ -41,14 +41,23 @@
-
+    private static final String CLIENT_ASSERTION_TYPE = "urn:ietf:params:oauth:client-assertion-type:jwt-bearer";
+    private static final String THUMBPRINT_ALGORITHM = "SHA-1";
+    private static final int ONE_HOUR = 60000;
```
Rename this value `ONE_MINUTE_MILLIS`

heidijinxujia commented:
I have addressed all the comments and going to merge this PR after Travis passing.

So far, I have spent two and half days last week and half day today in helping clean all these 700+ checkstyle warnings. I am not going to clean all the errors introduced by this morning's merged PRs. It will be endless work. 😢 

In the future, please make sure the checkstyle warnings is clean before merging by your PR. 

Thanks.

iambmelt commented:
@heidijinxujia Thanks for all of your hard work on these style checks. I'll pick up any remaining checkstyle issues introduced by #97 #103 or others. Sorry to overwhelm

iambmelt commented on config/checkstyle/suppressions.xml:
```diff
@@ -4,5 +4,13 @@
+  <suppress checks=".*" files=".*[\\/]unit[\\/]"/>
+  <suppress checks="JavadocVariable" files="MicrosoftStsIdToken.java"/>
+  <suppress checks="EmptyBlock" files="AzureActiveDirectoryOAuth2Strategy.java"/>
+  <suppress checks="EqualsHashCodeCheck" files="Account.java"/>
```
This should not have been merged without CR -- this is not the proper syntax to omit `equals()` and `hashCode()` from checkstyle

From the docs:
>Checks that classes that either override equals() or hashCode() also overrides the other. This checks only verifies that the method declarations match Object.equals(Object) and Object.hashCode() exactly to be considered an override. This check does not verify invalid method names, parameters other than Object, or anything else. <br /><br />Rationale: The contract of equals() and hashCode() requires that equal objects have the same hashCode. Hence, whenever you override equals() you must override hashCode() to ensure that your class can be used in collections that are hash based.

iambmelt commented on common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java:
```diff
@@ -63,13 +64,26 @@
+     * Constructor of ADALOAuth2TokenCache.
+     *
+     * @param context         Context
+     * @param sharedSSOCaches List<IShareSingleSignOnState>
```
Javadoc doesn't match the method declaration. This also should not have merged.
